source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -1070,23 +1040,18 @@ ShaderNode* ShaderGraph::createNode(const Node& node, GenContext& context)
     {
         for (InputPtr input : node.getInputs())
         {
-            ShaderInput* shaderInput = newNode->getInput(input->getName());
-            populateColorTransformMap(colorManagementSystem, shaderInput, input, targetColorSpace, true);
-        }
-        for (auto parameter : node.getParameters())
-        {
-            if (parameter->getType() == FILENAME_TYPE_STRING)
+            if (input->getIsUniform() && input->getType() == FILENAME_TYPE_STRING)
```

Comment:
File names should always be uniforms but perform a firewall check here to ensure previous logic..

Reply:
TO FIX: Will remove this since it's possible uniform is not set.